api/alerts.go

```diff
@@ -6,19 +6,24 @@ import (
 )
 
 type Alert struct {
-	ID                 string   `graphql:"id"                 yaml:"-"                     json:"id"`
-	Name               string   `graphql:"name"               yaml:"name"                  json:"name"`
-	QueryString        string   `graphql:"queryString"        yaml:"queryString"           json:"queryString"`
-	QueryStart         string   `graphql:"queryStart"         yaml:"queryStart"            json:"queryStart"`
-	ThrottleField      string   `graphql:"throttleField"      yaml:"throttleField"         json:"throttleField"`
-	TimeOfLastTrigger  int      `graphql:"timeOfLastTrigger"  yaml:"timeOfLastTrigger"     json:"timeOfLastTrigger"`
-	IsStarred          bool     `graphql:"isStarred"          yaml:"isStarred"             json:"isStarred"`
-	Description        string   `graphql:"description"        yaml:"description,omitempty" json:"description"`
-	ThrottleTimeMillis int      `graphql:"throttleTimeMillis" yaml:"throttleTimeMillis"    json:"throttleTimeMillis"`
-	Enabled            bool     `graphql:"enabled"            yaml:"enabled"               json:"enabled"`
-	Actions            []string `graphql:"actions"            yaml:"actions"               json:"actions"`
-	Labels             []string `graphql:"labels"             yaml:"labels,omitempty"      json:"labels,omitempty"`
-	LastError          string   `graphql:"lastError"          yaml:"lastError"             json:"lastError"`
+	ID                 string    `graphql:"id"                 yaml:"-"                     json:"id"`
+	Name               string    `graphql:"name"               yaml:"name"                  json:"name"`
+	QueryString        string    `graphql:"queryString"        yaml:"queryString"           json:"queryString"`
+	QueryStart         string    `graphql:"queryStart"         yaml:"queryStart"            json:"queryStart"`
+	ThrottleField      string    `graphql:"throttleField"      yaml:"throttleField"         json:"throttleField"`
+	TimeOfLastTrigger  int       `graphql:"timeOfLastTrigger"  yaml:"timeOfLastTrigger"     json:"timeOfLastTrigger"`
+	IsStarred          bool      `graphql:"isStarred"          yaml:"isStarred"             json:"isStarred"`
+	Description        string    `graphql:"description"        yaml:"description,omitempty" json:"description"`
+	ThrottleTimeMillis int       `graphql:"throttleTimeMillis" yaml:"throttleTimeMillis"    json:"throttleTimeMillis"`
+	Enabled            bool      `graphql:"enabled"            yaml:"enabled"               json:"enabled"`
+	Actions            []string  `graphql:"actions"            yaml:"actions"               json:"actions"`
+	Labels             []string  `graphql:"labels"             yaml:"labels,omitempty"      json:"labels,omitempty"`
+	LastError          string    `graphql:"lastError"          yaml:"lastError"             json:"lastError"`
+	RunAsUser          RunAsUser `graphql:"runAsUser"          yaml:"runAsUser"             json:"runAsUser"`
```

Comment:
When creating or updating, all other fields are of scalar types, but the run-as-user id, you need to wrap in an object. I would prefer, if that was not the case.

Also, note that for filter alerts, the run-as-user is not available in its own field in the output, there it is nested under the query owner. So you will have to be able to handle this field differently for input and output for filter alerts anyway, so I think it makes sense to do so already for alerts.

Reply:
We can have it unwrapped, however it will require 2 quite similar structs, one for unwrapping the graphql (the mutate method needs a struct in the same structure as the graphql - with the wrapper) and then one for the file format, and then map from and to when reading and writing files.

This "problem" is also present with queryOwnership which has a type and id:
```
queryOwnership {
  id,
  type: __typename,
}
```

Then the yaml file could look like:
```
queryOwnershipId: <id>
queryOwnershipType: <User | Organization>
```

And we could use the queryOwnershipId as out input to`CreateAlert( input: RunAsUserId: $queryOwnershipId...` when type is `User`.

Thoughts? It would separate the graphql format from the file format.

Reply:
I am mainly concerned about the API, not so much the CLI. I think that the {{Alert}} struct should have two fields, {{runAsUserId}} of type string and {{queryOwnershipType}} of type enum with two possible values. I do not think that we should have the query ownership id when the owner type is organization, so I do not like the name {{queryOwnerShipId}}.

In the yaml file, it should probably have two similar fields.

Reply:
I have seperated our alert struct from the graphql alert struct, and added ownershipType to this PR.